src/query/service/src/sessions/query_ctx.rs

```diff
@@ -502,52 +503,66 @@ impl TableContext for QueryContext {
         max_files: Option<usize>,
     ) -> Result<Vec<StageFileInfo>> {
         let tenant = self.get_tenant();
+        let mut files = files.clone();
         let catalog = self.get_catalog(catalog_name)?;
         let table = catalog
             .get_table(&tenant, database_name, table_name)
             .await?;
         let table_id = table.get_id();
 
         let mut limit: usize = 0;
+        let max_files = max_files.unwrap_or(MAX_QUERY_COPIED_FILES_NUM);
```

Comment:
https://github.com/datafuselabs/databend/pull/10790/files#diff-7c72c74465abbbfa3b0827662f229e83fa78c5473ad04f1cd4ca444d4088a47bR565

it is called with max_files=None?

Reply:
https://github.com/datafuselabs/databend/issues/10803 we can do it later